### DISCUSSION TITLE
Support for 'Infinity' value for timestamp (tz) on postgres

### DISCUSSION THREAD

**Discussion by renatocron** [directus/directus#4936] [Feature Requests]
I'm using directus:v9.0.0-rc.55 on docker, with postgresql version 12

Given this table (already existed, created without directus):

    CREATE TABLE public.config (
        id integer NOT NULL,
        name character varying NOT NULL,
        value character varying NOT NULL,
        valid_from timestamp without time zone DEFAULT now() NOT NULL,
        valid_to timestamp without time zone DEFAULT 'infinity'::timestamp without time zone NOT NULL
    );

and values like:

    id | name       |        value     |          valid_from           | valid_to
    11 | FOO        | bar                | 2021-04-02 10:10:56.336605-03 | infinity

After configuring the table (I set up both timestamp columns _Interface_ to _DateTime_, _Display_ to _Raw Value_) ¹, I was able to load the items after loading `/admin/collections/config`  when the `valid_to` column is not selected. If I select it, it returns the error after reloading:

    {
        "errors": [
          {
            "message": "Invalid time value",
            "extensions": {
              "code": "INTERNAL_SERVER_ERROR"
            }
          }
        ]
    }

and trying to load any item I get 404 "Page not found" error but the API returned 500, the node output is:

    22:57:41 🚨 Invalid time value 
    RangeError: Invalid time value
        at Object.formatISO (/directus/node_modules/date-fns/formatISO/index.js:62:11)
        at PayloadService.<anonymous> (/directus/node_modules/directus/dist/services/payload.js:227:57)
        at Generator.next (<anonymous>)
        at /directus/node_modules/directus/dist/services/payload.js:12:71
        at new Promise (<anonymous>)
        at __awaiter (/directus/node_modules/directus/dist/services/payload.js:8:12)
        at PayloadService.processDates (/directus/node_modules/directus/dist/services/payload.js:211:16)
        at PayloadService.<anonymous> (/directus/node_modules/directus/dist/services/payload.js:170:28)
        at Generator.next (<anonymous>)
        at fulfilled (/directus/node_modules/directus/dist/services/payload.js:9:58)
    22:57:41 ✨ request errored GET 500 /items/penhas_config/9 62ms

¹: after configuring it, it also changed the column to timestamp *with* time zone, which I think it's not correct to do without any warning, but it's fine in this table
 
I think it's important to support 'infinity' value for timestamp because this is relatively common pattern:

    CREATE UNIQUE INDEX idx_config_key ON public.config
         USING btree (name) WHERE (valid_to = 'infinity'::timestamp without time zone);

I also had to drop this index during setup, to be able to let directus change the column to timestamp with time zone, and then created it again.

**Comment by jaads** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
